src/nodes.coffee

```diff
@@ -1748,6 +1748,13 @@ exports.Assign = class Assign extends Base
     vvarText = fragmentsToText vvar
     assigns  = []
     expandedIdx = false
+    # check if variable is object and containes splat, e.g. {a, b, c...}
+    hasSplat = isObject and (i for obj, i in objects when obj instanceof Splat).length == 1
```

Comment:
Since multiple splats/expansions are disallowed in an assignment, length == 1 should be ok?